solr/core/src/java/org/apache/solr/handler/admin/api/BackupCoreAPI.java

```diff
@@ -0,0 +1,179 @@
+import io.swagger.v3.oas.annotations.parameters.RequestBody;
+import java.net.URI;
+import java.nio.file.Paths;
+import java.util.Optional;
+import javax.inject.Inject;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import org.apache.solr.common.SolrException;
+import org.apache.solr.common.params.CoreAdminParams;
+import org.apache.solr.core.CoreContainer;
+import org.apache.solr.core.SolrCore;
+import org.apache.solr.core.backup.BackupFilePaths;
+import org.apache.solr.core.backup.ShardBackupId;
+import org.apache.solr.core.backup.repository.BackupRepository;
+import org.apache.solr.handler.IncrementalShardBackup;
+import org.apache.solr.handler.SnapShooter;
+import org.apache.solr.handler.admin.CoreAdminHandler;
+import org.apache.solr.jersey.PermissionName;
+import org.apache.solr.jersey.SolrJerseyResponse;
+import org.apache.solr.request.SolrQueryRequest;
+import org.apache.solr.response.SolrQueryResponse;
+
+@Path(("/cores/{coreName}/backup/{name}"))
+public class BackupCoreAPI extends CoreAdminAPIBase {
+  @Inject
+  public BackupCoreAPI(
+      CoreContainer coreContainer,
+      SolrQueryRequest solrQueryRequest,
+      SolrQueryResponse solrQueryResponse,
+      CoreAdminHandler.CoreAdminAsyncTracker coreAdminAsyncTracker) {
+    super(coreContainer, coreAdminAsyncTracker, solrQueryRequest, solrQueryResponse);
+  }
+
+  @POST
+  @Produces({"application/json", "application/xml", BINARY_CONTENT_TYPE_V2})
+  @PermissionName(COLL_EDIT_PERM)
+  public SolrJerseyResponse createBackup(
+      @Schema(description = "The name of the core.") @PathParam("core") String coreName,
+      @Schema(description = "The backup will be created in a directory called snapshot.<name>")
+          @PathParam("name")
+          String name,
+      @Schema(description = "The POJO for representing additional backup params.") @RequestBody
+          BackupCoreRequestBody backupCoreRequestBody,
+      @Parameter(description = "The id to associate with the async task.") @QueryParam("async")
+          String taskId)
+      throws Exception {
+    if (coreName == null)
+      throw new SolrException(
+          SolrException.ErrorCode.BAD_REQUEST,
+          "Missing required parameter: " + CoreAdminParams.CORE);
+    return handlePotentiallyAsynchronousTask(
+        null,
+        coreName,
+        taskId,
+        "backup",
+        () -> {
+          try (BackupRepository repository =
+                  coreContainer.newBackupRepository(backupCoreRequestBody.repository);
+              SolrCore core = coreContainer.getCore(coreName)) {
+            String location = repository.getBackupLocation(backupCoreRequestBody.location);
+            if (location == null) {
+              throw new SolrException(
+                  SolrException.ErrorCode.BAD_REQUEST,
+                  "'location' is not specified as a query"
```

Comment:
[-0] Should probably update "query parameter" in this message to just read "parameter" - since `location` now appears in the request body of v2 requests.

Reply:
I was thinking of using the `JerseyResource.ensureRequiredParameterProvided` here as well. But looks like here we need more custom error message. I will just change the error message!